### PR TITLE
Fix incorrect name to TW

### PR DESCRIPTION
**Pull request by WilliamLINWEN**
Fix incorrect name to TW

### CODE REVIEW

**Comment by rposborne** [comment]
This is generally a contested change. And I have avoided accepting PR's involving Taiwan's name. I am in no way able to make this choice on my own.

File: lib/countries/data/countries/TW.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ TW:
   international_prefix: '002'
   ioc: TPE
   gec: TW
-  name: Taiwan, Province of China
+  name: Taiwan, Republic of china
```
Comment (jlnr) [inline lib/countries/data/countries/TW.yaml]: *China should be capitalized even here

**Comment by jlnr** [comment]
Some arguments for changing this entry:

* All the big players in IT call Taiwan just "Taiwan"; see e.g. [Google's region picker](https://store.google.com/de/countrypicker?hl=en-US), the macOS/iOS Region preferences, the country picker on dell.com, ...
* Even in [the recent series of back-and-forth about how Taiwan is called on websites](https://www.taiwannews.com.tw/en/news/3382136), websites flipped between "Taiwan" and "Taiwan, China" due to political pressure. No company seems to follow the ISO-3166 list.
* Postal mail sent to Taiwan is conventionally sent to "Taiwan, ROC" because otherwise [it can disappear](https://sentinel.tw/mail-not-arriving-taiwan/) – not something you want to have happen to users.

I think the right approach would be to include the unmodified ISO-3166 name, but *also* a property that identifies how countries are conventionally referred to on public-facing websites. People who use this gem through the `country_select` gem, for example, are not primarily looking for an ISO standard, but for an uncontroversial and unambiguous list of countries that will work on a website.

**Comment by jlnr** [comment]
OK, so later today I realized that the reason most companies list Taiwan as "Taiwan" is that they use the Unicode CLDR data, not ISO-3166. There's some good discussion about this in this [Drupal issue](https://www.drupal.org/project/drupal/issues/1938892).

Then I also noticed that #431 already mentions the `twitter-cldr` gem 🎉

**Comment by pmor** [comment]
As per https://www.iso.org/obp/ui/#iso:code:3166:TW, the official ISO 3166 name is "Taiwan, Province of China", so I'm closing this, same as I did #601. See my comment there for more information, but keep in mind the gem provides "common" names via the `translations`

**Pull request closed without merging**